core/src/main/java/com/google/cloud/sql/core/InternalConnectorRegistry.java

```diff
@@ -57,6 +57,9 @@ public final class InternalConnectorRegistry {
   private static InternalConnectorRegistry internalConnectorRegistry;
   private final ListenableFuture<KeyPair> localKeyPair;
   private final ConcurrentHashMap<ConnectorKey, Connector> connectors = new ConcurrentHashMap<>();
+  private final ConcurrentHashMap<String, ConnectionConfig> namedConfigs =
+      new ConcurrentHashMap<>();
+  private final ConcurrentHashMap<String, Connector> namedConnectors = new ConcurrentHashMap<>();
```

Comment:
Could we update ConnectorKey to account for connector name and thereby combine this map with the `connectors` map above?